### PR TITLE
Added enrollment percentage per country to enrollment geography table.

### PR DESCRIPTION
**Pull request by dsjen**
percentages

@rocha @clintonb

### CODE REVIEW

File: analytics_dashboard/courses/presenters.py [hunk 1]
```diff
@@ -106,10 +106,15 @@ def get_geography_data(self):
 
         if api_response:
             update_date = api_response[0]['date']
+
+            # get the sum as a float so we can divide by it to get a percent
+            sum_count = float(sum([datum['count'] for datum in api_response]))
```
Comment (clintonb) [inline analytics_dashboard/courses/presenters.py]: Can you use a more descriptive variable name such as `total_enrollment`?

File: analytics_dashboard/static/js/views/data-table-view.js [hunk 2]
```diff
@@ -76,8 +89,6 @@ define(['dataTablesBootstrap', 'jquery', 'underscore', 'utils/utils', 'views/att
                 var self = this,
                     $parent = $('<div/>', {class:'table-responsive'}).appendTo(self.$el),
                     $table = $('<table/>', {class: 'table table-striped'}).appendTo($parent),
-                    $thead = $('<thead/>').appendTo($table),
```
Comment (clintonb) [inline analytics_dashboard/static/js/views/data-table-view.js:79]: How are headers specified without `<thead>`?

File: analytics_dashboard/courses/presenters.py [hunk 3]
```diff
@@ -106,10 +106,15 @@ def get_geography_data(self):
...
+
             # formatting this data for easy access in the table UI
             data = [{'countryCode': datum['country']['alpha3'],
                      'countryName': datum['country']['name'],
-                     'count': datum['count']} for datum in api_response]
+                     'count': datum['count'],
+                     'percent': datum['count'] / sum_count if sum_count > 0 else 0.0} for datum in api_response]
```
Comment (clintonb) [inline analytics_dashboard/courses/presenters.py]: If total enrollment is 0, and this block of code is getting executed, we have some major issues. Probably not something to resolve here, but do we have designs/implementations for empty states (e.g. courses with no enrollment)?

**Comment by clintonb** [comment]
Looks fine. Just rename `sum_count`. :+1:

File: analytics_dashboard/courses/presenters.py (hunk 1 above)
Comment (dsjen) [inline analytics_dashboard/courses/presenters.py]: That makes sense.  Can do!

File: analytics_dashboard/courses/presenters.py (hunk 3 above)
Comment (dsjen) [inline analytics_dashboard/courses/presenters.py]: Yes, we should definitely figure out the states when there's no data.  I thought I created a task for it, but can't find it now.

File: analytics_dashboard/static/js/views/data-table-view.js (hunk 2 above)
Comment (dsjen) [inline analytics_dashboard/static/js/views/data-table-view.js:79]: It's the magic of datatables!  Specifying the "title" attribute in the columndefs will populate the header.  I'll add a comment about that.

Comment (clintonb) [inline analytics_dashboard/static/js/views/data-table-view.js:79]: Does the "magic" render a `<thead>`? There may be accessibility issues here. Please hold off on merging this until after our architecture meeting.

Comment (dsjen) [inline analytics_dashboard/static/js/views/data-table-view.js:79]: I checked and it does render `<thead>`, and I'm happy to wait until architecture meeting.

Comment (clintonb) [inline analytics_dashboard/static/js/views/data-table-view.js:79]: In that case, it's fine. We'll be discussing accessibility today, so some changes may be needed. Carry on.

File: analytics_dashboard/static/js/views/data-table-view.js [hunk 4]
```diff
@@ -49,24 +33,54 @@ define(['dataTablesBootstrap', 'jquery', 'underscore', 'utils/utils', 'views/att
...
+                                display = value;
+                            if (type === 'display') {
+                                display = ' < 1%';
+                                if (value >= 0.01) {
+                                    display = (value * 100).toFixed(1) + '%';
+                                }
+                            }
+                            return display;
+                        };
                     }
```
Comment (rocha) [inline analytics_dashboard/static/js/views/data-table-view.js:62]: Minor: I find a lot easier to read when longish functions like this one are refactored as:

```
if (column.type === 'date') {
  def.data = format_date_data(args..)
} else if (column.type === 'percent') {
  def.data = format_percent_data(args...)
}
```

**Comment by rocha** [comment]
:rocket: (I need to learn how datatable works, lots of magic there)

File: analytics_dashboard/static/js/views/data-table-view.js (hunk 4 above)
Comment (dsjen) [inline analytics_dashboard/static/js/views/data-table-view.js:62]: I went ahead and refactored it (seems like a good pattern to follow).  Please let me know if it looks ok for a merge.

Comment (rocha) [inline analytics_dashboard/static/js/views/data-table-view.js:62]: Oh yeah, LGTM :rocket: . It was a minor comment, so is up to you if you want to address it or not :smile:

**Pull request merged**